### DISCUSSION TITLE
esp32 MIDI Out Problem

### DISCUSSION THREAD

**Discussion by craiglindley** [FortySevenEffects/arduino_midi_library#312] [General]
This is the first time I have used this library so I may be missing something but I cannot get MIDI output to work at all with Serial2. The code I am using is shown below. On the MIDI DIN connector I connected the ground pin to my circuit ground, I connected GPIO4 to pin 5 through a 10 ohm resistor and pin 4 to 3.3V through a 33 ohm resistor.

I have tried 2 different ESP32 modules and three different GPIO pins and nothing seems to work. I also tried to output MIDI to the main serial port and could see garbage chars written to the serial monitor every second which I interpret to mean that MIDI data is being written out but when I change back to Serial2 none of my connected MIDI devices respond. 

`#include <MIDI.h>

// MIDI output signal
#define MIDI_OUT_PIN 4

#include <MIDI.h>
#include <HardwareSerial.h>

// Override default MIDI settings
struct Serial2MIDISettings : public midi::DefaultSettings {
  static const long BaudRate = 31250;
  static const int8_t TxPin = MIDI_OUT_PIN;
};

/************************************************************************/
/***                         Driver Instances                         ***/
/************************************************************************/

// Create MIDI driver instance called MIDI
MIDI_CREATE_CUSTOM_INSTANCE(HardwareSerial, Serial2, MIDI, Serial2MIDISettings);

void setup() {

  MIDI.begin();
}

void loop() {
  MIDI.sendNoteOn(60, 64, 1);
  delay(1000);
  MIDI.sendNoteOff(60, 64, 1);
  delay(1000);
}
`
Any suggestions would be appreciated. Thanks

**Comment by lathoub** [comment]
There is no need to override the midi::DefaultSettings (Baudrate is already set to the default 31250 and setting the TXPin in Settings does not make sense, it will not be used by the library). 

Use the regular `MIDI_CREATE_INSTANCE(HardwareSerial, Serial2,  MIDI);` macro.

Do  a google search for ESP32 and MIDI for the schema, resistor values and the optocoupler. (eg https://github.com/alf45tar/PedalinoMini/issues/12)
